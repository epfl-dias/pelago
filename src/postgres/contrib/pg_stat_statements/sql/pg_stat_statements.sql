CREATE EXTENSION pg_stat_statements;

CREATE TABLE test (a int, b char(20));

-- test the basic functionality of pg_stat_statements
SELECT pg_stat_statements_reset();

INSERT INTO test VALUES(generate_series(1, 10), 'aaa');
UPDATE test SET b = 'bbb' WHERE a > 5;

SELECT query, calls, rows from pg_stat_statements ORDER BY rows;

DROP TABLE test;

DROP EXTENSION pg_stat_statements;
