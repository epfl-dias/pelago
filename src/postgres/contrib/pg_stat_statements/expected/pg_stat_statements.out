CREATE EXTENSION pg_stat_statements;
CREATE TABLE test (a int, b char(20));
-- test the basic functionality of pg_stat_statements
SELECT pg_stat_statements_reset();
 pg_stat_statements_reset 
--------------------------
 
(1 row)

INSERT INTO test VALUES(generate_series(1, 10), 'aaa');
UPDATE test SET b = 'bbb' WHERE a > 5;
SELECT query, calls, rows from pg_stat_statements ORDER BY rows;
                       query                        | calls | rows 
----------------------------------------------------+-------+------
 SELECT pg_stat_statements_reset();                 |     1 |    1
 UPDATE test SET b = ? WHERE a > ?;                 |     1 |    5
 INSERT INTO test VALUES(generate_series(?, ?), ?); |     1 |   10
(3 rows)

DROP TABLE test;
DROP EXTENSION pg_stat_statements;
